src/Endpoint.js

```diff
@@ -60,15 +59,11 @@ class Endpoint<P, T> {
       return new Payload(false, url, this.responseOverride, null)
     }
 
-    let response = null
-    try {
-      response = await (this.mapParamsToBody ? this.postFormData(url, params) : fetch(url))
-    } catch (e) {
-      throw new FetchError({ endpointName: this.stateName, innerError: e })
-    }
+    const formData = this.mapParamsToBody ? this.mapParamsToBody(params) : null
+    const response = await this.fetchOrPost(url, formData)
 
     if (!response.ok) {
-      throw new ResponseError({ endpointName: this.stateName, response })
+      throw new ResponseError({ endpointName: this.stateName, responseStatus: response.status, url, formData })
```

Comment:
Was having `response` in the error not helpful? I guess we could extract the server's response (i.e. in text form) from that. Or is there another reason you replaced it with response.status?